### ISSUE TITLE
numpy.asarray() removes fill value from Pandas SparseArrays

### ISSUE THREAD

**Issue by jtanman** [pandas-dev/pandas#29231]
#### Code Sample, a copy-pastable example if possible

```python
import numpy as np
import pandas as pd

arr = np.random.randint(2, size=10)
arr
array([0, 1, 0, 1, 0, 0, 0, 0, 0, 1])

sparr = pd.SparseArray(arr, fill_value=0)
sparr
[0, 1, 0, 1, 0, 0, 0, 0, 0, 1]
Fill: 0
IntIndex
Indices: array([1, 3, 9], dtype=int32)

np_arr = np.asarray(sparr)
np_arr
array([1, 1, 1])
```
#### Problem description

Converting back to a dense array from a Pandas SparseArary no longer works as intended. It should keep all values in the array as is, but instead the new array removes all elements that are the fill_value. Thus the new array ends up with less elements than before.

**Note**: We receive a lot of issues on our GitHub tracker, so it is very possible that your issue has been posted before. Please check first before submitting so that we do not have to handle and close duplicates!

**Note**: Many problems can be resolved by simply upgrading `pandas` to the latest version. Before submitting, please check if that solution works for you. If possible, you may want to check if `master` addresses this issue, but that is not necessary.

For documentation-related issues, you can check the latest versions of the docs on `master` here:

https://pandas-docs.github.io/pandas-docs-travis/

If the issue has not been resolved there, go ahead and file it in the issue tracker.

#### Expected Output

```python
np_arr
array([0, 1, 0, 1, 0, 0, 0, 0, 0, 1])
```

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: None
python: 3.7.1.final.0
python-bits: 64
OS: Darwin
OS-release: 18.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.23.4
pytest: 5.2.1
pip: 18.1
setuptools: 41.4.0
Cython: 0.29.13
numpy: 1.17.3
scipy: 1.3.1
pyarrow: None
xarray: None
IPython: 7.8.0
sphinx: None
patsy: 0.5.1
dateutil: 2.8.0
pytz: 2019.3
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 3.1.1
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: 1.3.10
pymysql: 0.9.3
psycopg2: None
jinja2: 2.10.3
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

**Comment by hwalinga** [comment]
I cannot reproduce this with the latest release. So, updating you pandas package will fix this.

**Comment by TomAugspurger** [comment]
Fixed in 0.25 I think.
